File: Android/app/src/main/java/com/bdreiss/trackmyattack/listeners/RemedyOnClickListener.java
package com.bdreiss.trackmyattack.listeners;

import android.content.Context;
import android.view.View;

import main.java.com.bdreiss.dataAPI.DataModel;
import main.java.com.bdreiss.dataAPI.EntryNotFoundException;
import main.java.com.bdreiss.dataAPI.Intensity;
import main.java.com.bdreiss.dataAPI.IteratorWithIntensity;
import main.java.com.bdreiss.dataAPI.TypeMismatchException;

public class RemedyOnClickListener extends CustomListener{

    public RemedyOnClickListener(Context context, DataModel data){
        super(context, data);
    }
    @Override
    public void onClick(View view) {
        super.onClick(view);

        try {
            if (data.getRemedyData(text) instanceof IteratorWithIntensity){
                chooseIntensity(context, (dialogInterface, i) -> {
                    try {
                        data.addRemedy(text, Intensity.values()[i+1]);
                    } catch (TypeMismatchException e) {
                        e.printStackTrace();
                    }

                });
            }
            else{
                data.addRemedy(text);

            }
        } catch (EntryNotFoundException | TypeMismatchException e) {
            e.printStackTrace();
        }

    }

    @Override
    public CustomListener copy() {
        RemedyOnClickListener newListener = new RemedyOnClickListener(context, data);
        newListener.setTextValue(text);
        return newListener;
    }
}

File: Android/app/src/main/java/com/bdreiss/trackmyattack/listeners/RemedyOnClickListener.kt
package com.bdreiss.trackmyattack.listeners

import android.content.Context
import android.view.View
import main.java.com.bdreiss.dataAPI.DataModel
import main.java.com.bdreiss.dataAPI.IteratorWithIntensity

//Listener for adding data to remedies
class RemedyOnClickListener(context: Context, data: DataModel) : CustomListener(context, data) {

    override fun onClick(v: View?) {

        if (data.getRemedyData(text) is IteratorWithIntensity)
            chooseIntensity(context){
                    chosenEnumValue ->
                data.addRemedy(text, chosenEnumValue)
                data.save()
            }
        else
            data.addRemedy(text)
        data.save()

    }

    override fun clone(): CustomListener {
        return RemedyOnClickListener(context, data)
    }

}
